pkg/sortition/sortition.go

```diff
@@ -0,0 +1,118 @@
+package sortition
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ipfs/go-log"
+)
+
+const (
+	DefaultStatusCheckTick = 10 * time.Second
+)
+
+var logger = log.Logger("keep-sortition")
+
+var errOperatorUnknown = fmt.Errorf("operator not registered for the staking provider, check Threshold dashboard")
+
+// MonitorPool periodically checks the status of the operator in the sortition
+// pool. If the operator is supposed to be in the sortition pool but is not
+// there yet, the function attempts to add the operator to the pool. If the
+// operator is already in the pool and its status is no longer up to date, the
+// function attempts to update the operator's status in the pool.
+func MonitorPool(
+	ctx context.Context,
+	chain Chain,
+	tick time.Duration,
+) error {
+	_, isRegistered, err := chain.OperatorToStakingProvider()
+	if err != nil {
+		return fmt.Errorf("could not resolve staking provider: [%v]", err)
+	}
+
+	if !isRegistered {
+		return errOperatorUnknown
+	}
+
+	err = checkOperatorStatus(chain)
+	if err != nil {
+		logger.Errorf("could not check operator sortition pool status: [%v]", err)
+	}
+
+	ticker := time.NewTicker(tick)
+
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				ticker.Stop()
+				return
+			case <-ticker.C:
+				err = checkOperatorStatus(chain)
+				if err != nil {
+					logger.Errorf("could not check operator sortition pool status: [%v]", err)
+					continue
+				}
+			}
+		}
+	}()
+
+	return nil
+}
+
+func checkOperatorStatus(chain Chain) error {
+	logger.Info("checking sortition pool operator status")
+
+	isOperatorInPool, err := chain.IsOperatorInPool()
+	if err != nil {
+		return err
+	}
+
+	if isOperatorInPool {
+		logger.Info("operator is in the sortition pool")
+	} else {
+		logger.Info("operator is not in the sortition pool")
+	}
+
+	isOperatorUpToDate, err := chain.IsOperatorUpToDate()
+	if err != nil {
+		return err
+	}
+
+	if isOperatorUpToDate {
+		if isOperatorInPool {
+			logger.Infof("sortition pool operator status is up to date")
```

Comment:
I think we can use just `logger.Info`, not `logger.Infof` cause there's nothing to format here. Same comment for other places like this in this file.

Reply:
Yeah, good point. Let's fix it when working on #3008.